src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskTest.java

```diff
@@ -413,6 +416,52 @@ void testSourceLoadsDataFromCollectionDocumentOnly() {
     }
   }
 
+  @Test
+  @DisplayName("Ensure source sends data to the deadletter queue on failures")
```

Comment:
Is there a test for errorTolerance of ALL and no dead letter queue configured?  If not, let's add one.

Reply:
That will come as part of KAFKA-105